### PR TITLE
#8 Auto-dataset via custom runner

### LINKED ISSUE #8: Investigate overriding default MemoryDataSet
Right now the plug-in requires all intermediate datasets to be stored in GCS, which forces the pipeline authors to remember that all intermediate datasets need to be explicitly defined in the data catalog.
Consider overriding the `create_default_data_set` in the `ThreadRunner` https://github.com/kedro-org/kedro/blob/74fbd752b6bf77c409f016ed73e60a4ea38d6a95/kedro/runner/thread_runner.py#L51 or find other extension point to make the use of intermediate datasets transparent to the users (we could store them in GCS, as we already have paths to bucket and unique run id in the plugin).

### PR DESCRIPTION
**Pull request by marrrcin**
#### Description

Back port auto-generating dataset from kedro-azureml to Vertex AI Pipelines. Thanks to this change, user is not forced to add all of the intermediate datasets to the catalog. 

Resolves #8 

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: docs/source/03_getting_started/01_quickstart.md [hunk 1]
```diff
@@ -119,117 +119,14 @@ shuttles:
...
-data_science.candidate_modelling_pipeline.y_test:
-  type: pickle.PickleDataSet
-  filepath: gs://STAGING_BUCKET/${run_id}/05_model_input/y_test.pickle
-  layer: model_input
-
-data_science.candidate_modelling_pipeline.regressor:
-  type: pickle.PickleDataSet
-  filepath: gs://STAGING_BUCKET/${run_id}/06_models/regressor.pickle
-  versioned: true
-  layer: models
```
Comment (szczeles) [inline docs/source/03_getting_started/01_quickstart.md:228]: :heart_eyes:

File: kedro_vertexai/cli.py [hunk 2]
```diff
@@ -99,16 +99,21 @@ def run_once(
...
+    run = client.run_once(
         pipeline=pipeline,
         image=image if image else config.image,
         image_pull_policy=config.image_pull_policy,
         parameters=format_params(params),
     )
 
+    click.echo(
+        f"Intermediate data datasets will be stored in{os.linesep}"
+        f"gs://{config.root.strip('/')}/{KEDRO_VERTEXAI_BLOB_TEMP_DIR_NAME}/{run['displayName']}/*.bin"
```
Comment (szczeles) [inline kedro_vertexai/cli.py:114]: `run['displayName']` is used in the message, but files land in `run.run_id` subdir - are these same?

**Review by szczeles — APPROVED** [review]
Amazing improvement!

File: kedro_vertexai/cli.py (hunk 2 above)
Comment (marrrcin) [reply kedro_vertexai/cli.py:114]: Actually, display name is used here (https://github.com/getindata/kedro-vertexai/pull/63/files#diff-efca27e2c6efab2c456a964844cf6ce99c04171ab0640a66ed358bd18a633a66R49), but indeed - the parameter in the `KedroVertexAIRunnerDataset` is named `run_id`, which is misleading. I will rename it to `unique_id`.

**Pull request merged**